src/plans/vectorial_plan.jl

```diff
@@ -0,0 +1,520 @@
+using its component functions,
+
+```math
+f(p) = \bigl( f_1(p), f_2(p), \ldots, f_n(p) \bigr)^{\mathrm{T}}
+```
+
+In this representation `f` is given as a vector `[f1(M,p), f2(M,p), ..., fn(M,p)]`
+of its component functions.
+An advantage is that the single components can be evaluated and from this representation
+one even can directly read of the number `n`. A disadvantage might be, that one has to
+implement a lot of individual (component) functions.
+
+For the  [`FunctionVectorialType`](@ref) ``f`` is implemented as a single function
+`f(M, p)`, that returns an `AbstractArray`.
+And advantage here is, that this is a single function. A disadvantage might be,
+that if this is expensive even to compute a single component, all of `f` has to be evaluated
+
+For the [`ComponentVectorialType`](@ref) of `f`, each of the component functions
+is a (classical) objective and hence has a gradient ``\oepratorname{grad} f_i(p) ∈ T_p\mathcal M``.
+Putting these gradients into a vector the same way as the functions, we obtain a
+[`ComponentVectorialType`](@ref)
+
+```math
+\operatorname{grad} f(p) = \Bigl( \operatorname{grad} f_1(p), \operatorname{grad} f_2(p), …, \operatorname{grad} f_n(p) \Bigr)^{\mathrm{T}}
+∈ (T_p\mathcal M)^n
+```
+
+And advantage here is, that again the single components can be evaluated individually
+
+# Fields
+
+* `value!!`:          the cost function ``f``, which can take different formats
+* `cost_type`:     indicating / string data for the type of `f`
+* `jacobian!!:     the jacobian of ``f``
+* `jacobian_type`: indicating / storing data for the type of ``J_f``
+* `parameters`:    the nunmber `n` from above, that is the size of the vector ``f`` returns.
+
+# Constructor
+
+VectorGradientFunction(f, Jf, range_dimension;
+    evaluation::AbstractEvaluationType=AllocatingEvaluation(),
+    function_type::AbstractVectorialType=FunctionVectorialType(),
+    jacobian_type::AbstractVectorialType=FunctionVectorialType(),
+)
+"""
+struct VectorGradientFunction{
+    E<:AbstractEvaluationType,
+    FT<:AbstractVectorialType,
+    JT<:AbstractVectorialType,
+    F,
+    J,
+    I<:Integer,
+} <: Function
+    value!!::F
+    cost_type::FT
+    jacobian!!::J
+    jacobian_type::JT
+    range_dimension::I
+end
+
+function VectorGradientFunction(
+    f::F,
+    Jf::J,
+    range_dimension::I;
+    evaluation::E=AllocatingEvaluation(),
+    function_type::FT=FunctionVectorialType(),
+    jacobian_type::JT=FunctionVectorialType(),
+) where {
+    I,F,J,E<:AbstractEvaluationType,FT<:AbstractVectorialType,JT<:AbstractVectorialType
+}
+    return VectorGradientFunction{E,FT,JT,F,J,I}(
+        f, function_type, Jf, jacobian_type, range_dimension
+    )
+end
+
+@doc raw"""
+    get_value(M::AbstractManifold, vgf::VectorGradientFunction, p, i, range=nothing)
+
+Evaluate the vector function [`VectorGradientFunction`](@ref) `vgf` at at `p`.
+The `range` can be used to speficy a potential range, but is currently only present for consistency.
```

Comment:
I don't think we need an unused argument for consistency?

Reply:
sure I also think by now this will stay in R^n anyways, so then we do not need that. Will remove it in the next commit.